app/screens/convert_gm_to_channel/team_selector.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Platform} from 'react-native';
+
+import OptionItem from '@app/components/option_item';
+import {Screens} from '@app/constants';
+import {useTheme} from '@app/context/theme';
+import {dismissBottomSheet, goToScreen} from '@app/screens/navigation';
+import {preventDoubleTap} from '@app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@app/utils/theme';
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        teamSelector: {
+            borderTopWidth: 1,
+            borderBottomWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+    };
+});
+
+type Props = {
+    commonTeams: Team[];
+    onSelectTeam: (team: Team) => void;
+    selectedTeamId?: string;
+}
+
+export const TeamSelector = ({commonTeams, onSelectTeam, selectedTeamId}: Props) => {
+    const {formatMessage} = useIntl();
+    const theme = useTheme();
+    const styles = getStyleFromTheme(theme);
+
+    const [selectedTeam, setSelectedTeam] = useState<Team>();
+
+    const label = formatMessage({id: 'channel_into.convert_gm_to_channel.team_selector.label', defaultMessage: 'Team'});
+    const placeholder = formatMessage({id: 'channel_into.convert_gm_to_channel.team_selector.placeholder', defaultMessage: 'Select a Team'});
+
+    useEffect(() => {
+        if (selectedTeamId) {
+            const team = commonTeams.find((t) => t.id === selectedTeamId);
+            setSelectedTeam(team);
+        }
+    }, [selectedTeamId]);
+
+    const selectTeam = useCallback((teamId: string) => {
+        const team = commonTeams.find((t) => t.id === teamId);
+        if (team) {
+            setSelectedTeam(team);
+            onSelectTeam(team);
```

Comment:
this will cause the previous `useEffect` to trigger again as the prop for `selectedTeamId` should change on the parent component and  I believe and the value is already set on L51, so is being set "twice".. not major but just curious if you can prevent that.

Reply:
I updated the previous useEffect to only set if the state isn't already set. This prevents the state being set twice.